### PR TITLE
storage: optimize the storage interface

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: HunDunDM <hundundm@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #3453

### What is changed and how does it work?

Currently, in the interface definition of `RegionStorage`, `LoadRegions` and `LoadRegionsOnce` are repeated, and the implementation of the `LoadRegionsOnce` is not always consistent with the definition. In addition, some interface conversions based on `coreStorage` are tricky.

```commit-message
storage: optimize the storage interface
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — COMMENTED** [review]
Basically LGTM. Would you like to add some test cases for the newly added function `TrySwitchRegionStorage`?

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
> Basically LGTM. Would you like to add some test cases for the newly added function `TrySwitchRegionStorage`?

cc @HunDunDM

**Review by AndreMouche — APPROVED** [review]
LGTM

File: server/storage/storage.go [hunk 1]
```diff
@@ -135,6 +139,22 @@ func (ps *coreStorage) LoadRegionsOnce(ctx context.Context, f func(region *core.
 	return nil
 }
 
+// LoadRegion loads one region from storage.
+func (ps *coreStorage) LoadRegion(regionID uint64, region *metapb.Region) (ok bool, err error) {
```
Comment (rleungx) [inline server/storage/storage.go:143]: Seem this function only be used in the tests

**Comment by nolouch** [comment]
ptal @HunDunDM

**Review by lhy1024 — APPROVED** [review]

File: server/storage/storage.go (hunk 1 above)
Comment (HunDunDM) [reply server/storage/storage.go:143]: It is an implementation of interface.

Comment (rleungx) [reply server/storage/storage.go:143]: Yes, but it's only used for tests.

**Pull request merged**